src/cloudant/database.py

```diff
@@ -485,48 +491,112 @@ def unshare_database(self, username):
-            )
+        )
         resp.raise_for_status()
         return resp.json()
 
     def shards(self):
         """
-        GET /db/_shards Returns information about the shards in a database or the shard a document belongs to
+        _shards_
+
+        Returns information about the shards in a database
 
         """
-        pass
+        url = posixpath.join(self.database_url, '_shards')
+        resp = self._r_session.get(url)
+        resp.raise_for_status()
+
+        return resp.json()
 
-    def missing_revisions(self):
+    def missing_revisions(self, doc_id, *revisions):
         """
-        POST    /db/_missing_revs   Given a list of document revisions, returns the document revisions that do not exist in the database
+        _missing_revisions_
+
+        Given a document id and list of document revisions, returns the
+          document revisions that do not exist in the database
+
+        :param doc_id: document _id to check for missing revisions on
+        :param revisions: document _revs to check
 
         """
-        pass
+        url = posixpath.join(self.database_url, '_missing_revs')
+        data = {doc_id: list(revisions)}
 
-    def revisions_diff(self, *revisions):
+        resp = self._r_session.post(
+            url,
+            headers={'Content-Type': 'application/json'},
+            data=json.dumps(data)
+        )
+        resp.raise_for_status()
+
+        resp_json = resp.json()
+        missed_revs = resp_json['missed_revs'][doc_id]
+
+        return missed_revs
+
+    def revisions_diff(self, doc_id, *revisions):
         """
-        POST    /db/_revs_diff  Given a list of document revisions, returns differences between the given revisions and ones that are in the database
+        _revisions_diff_
+
+        Given a list of document revisions, returns differences between the
+          given revisions and ones that are in the database
+
+        :param doc_id: document _id to check for missing revisions on
+        :param revisions: document _revs to check
 
         """
-        pass
+        url = posixpath.join(self.database_url, '_revs_diff')
+        data = {doc_id: list(revisions)}
 
-    def get_revision_limit(self, doc):
+        resp = self._r_session.post(
+            url,
+            headers={'Content-Type': 'application/json'},
+            data=json.dumps(data)
+        )
+        resp.raise_for_status()
+
+        return resp.json()
+
+    def get_revision_limit(self):
         """
-        GET /db/_revs_limit Gets the limit of historical revisions to store for a single document in the database
+        _get_revision_limit_
+
+        Gets the limit of historical revisions to store for a single document
+          in the database
 
         """
-        pass
+        url = posixpath.join(self.database_url, '_revs_limit')
+        resp = self._r_session.get(url)
+        resp.raise_for_status()
+
+        return resp.text
```

Comment:
response is an int, didn't know if json() would render properly and was lazy and didn't test it out, but just did and resp.json() renders properly so I'll change that for consistency